model/RolePrivilegeRetriever.php

```diff
@@ -58,6 +57,15 @@ public function retrieveByResourceIds(array $resourceIds): array
             $permissions[$user][] = $result['privilege'];
         }
 
+        // Remove possible duplicates caused by merging ACLs from different
+        // resources: We picked the ACLs and merged them together by user (i.e.
+        // discarding the resource ID), but we've not checked for duplicates, so
+        // we need to filter them here.
+        //
+        foreach ($permissions as $_roleURI => &$entries) {
+            $entries = array_unique($entries);
```

Comment:
Some points. 

- I think instead of writing a big explanation, maybe create a private method called `removeDuplicatedEntries($entries)` can do the same?
- Also, keep in mind that array_unique does not work with matrix https://www.php.net/manual/en/function.array-unique.php#refsect1-function.array-unique-notes

Reply:
In this case (in this class), `$entries` is an array of strings (an array of grants). For example,

```
$permissions = [
     'http://www.tao.lu/Ontologies/TAO.rdf#BackOfficeRole' => ['GRANT', 'READ', 'WRITE']
]

// $entries is the array of strings
$entries = ['GRANT', 'READ', 'WRITE']
```
For clarity, it would be better to have some sort of value object to hold the `role => [permission1, permission2]` associations but that would bring even more noise to this PR :S).

Reply:
- [ ] Move the foreach to a private method

Reply:
b77fba36e925151f993daa8f03138dd8d6a985e5